res/controllers/Stanton-DJC-4-scripts.js

```diff
@@ -0,0 +1,359 @@
+/**
+ * Stanton DJC4 controller script v1.0 for Mixxx v2.2.3
+ *
+ * Written by Martin Bruset Solberg
+ * Adopted for v2.2.3 by Christoph Zimmermann
+ *
+ * Based on:
+ *  Denon MC2000 script by Esteban Serrano Roloff,
+ *  Denon MC7000 script by OsZ
+ *  Roland DJ-505 script by Jan Holthuis
+ *
+ * TODO:
+ * Pitch range
+ *
+ **/
+
+var djc4 = {};
+
+/////////////////
+// Tweakables. //
+/////////////////
+
+djc4.tempoRange = [0.08, 0.16, 0.5];
+djc4.autoShowFourDecks = false;
+
+///////////
+// Code. //
+///////////
+
+// ----------   Global variables    ----------
+
+// MIDI Reception commands (from spec)
+djc4.leds = {
+    loopminus: 2,
+    loopplus: 3,
+    loopin: 4,
+    loopout: 5,
+    loopon: 6,
+    loopdel: 7,
+    hotcue1: 8,
+    hotcue2: 9,
+    hotcue3: 10,
+    hotcue4: 11,
+    sample1: 12,
+    sample2: 13,
+    sample3: 14,
+    sample4: 15,
+    keylock: 16,
+    sync: 18,
+    pbendminus: 19,
+    pbendplus: 20,
+    scratch: 21,
+    tap: 22,
+    cue: 23,
+    play: 24,
+    highkill: 25,
+    midkill: 26,
+    lowkill: 27,
+    pfl: 28,
+    fxon: 30,
+    fxexf1: 31,
+    fxexf2: 32,
+    fxexf3: 33,
+    loadac: 34,
+    loadbd: 35,
+    videofx: 36,
+    xflink: 37,
+    keyon: 38,
+    filteron: 39,
+    tx: 46,
+    fx: 47
+};
+
+// ----------   Functions    ----------
+
+// Called when the MIDI device is opened & set up.
+djc4.init = function() {
+    var i;
+
+    // Put all LEDs to default state.
+    djc4.allLed2Default();
+
+    engine.makeConnection("[Channel3]", "track_loaded", djc4.autoShowDecks);
+    engine.makeConnection("[Channel4]", "track_loaded", djc4.autoShowDecks);
+
+    if (engine.getValue("[Master]", "num_samplers") < 8) {
+        engine.setValue("[Master]", "num_samplers", 8);
+    }
```

Comment:
Can explain more in depth why this is done? It seems to me like this has nothing to do with the controller itself.

Reply:
I copied this from the Reloop Terminal Mix2-4 mapping. The controller mapping contains 8 sampler buttons. This makes sure that all buttons get actually something to work with, to avoid confusion when people forget to increase this value by hand.

You are right, that this influences the Mixxx user settings which is not favorable, this is why I gets just increased if it is lower else it's untouched.

Reply:
I just looked up the controller and since it has eight samplers which are accessible all the time, it makes sense to keep this code.